src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenLocalFunctionTests.cs

```diff
@@ -31,6 +31,41 @@ public static IMethodSymbol FindLocalFunction(this CommonTestBase.CompilationVer
     public class CodeGenLocalFunctionTests : CSharpTestBase
     {
         [Fact]
+        [WorkItem(16399, "https://github.com/dotnet/roslyn/issues/16399")]
+        public void RecursiveGenericLocalFunctionIterator()
+        {
+            var src = @"
+using System;
+using System.Collections.Generic;
+using System.Linq;
+public static class EnumerableExtensions
+{
+    static void Main(string[] args)
+    {
+        GetLeaves<object>(new List<object>(), list => null);
+    }
+    public static IEnumerable<T> GetLeaves<T>(T root, Func<T, IEnumerable<T>> getChildren)
+    {
+        return GetLeaves(root);
+
+        IEnumerable<T> GetLeaves(T node)
+        {
+            var children = getChildren(node);
+            if (children == null)
+            {
+                return null;
+            }
+            else
+            {
+                return children.SelectMany(GetLeaves);
+            }
+        }
+    }
+}";
+            VerifyOutput(src, "");
```

Comment:
>VerifyOutput(src, ""); [](start = 12, length = 22)

Consider changing the test and the input so that it would output something to confirm that the behavior of the compiled assembly is correct.